plugins/woocommerce/client/admin/client/wp-admin-scripts/email-editor-integration/blocks/woo-email-content/placeholder-block-edit-content.tsx

```diff
@@ -0,0 +1,127 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState, useEffect, useRef } from '@wordpress/element';
+import { useEntityRecord, Post } from '@wordpress/core-data';
+import { EntityRecordResolution } from '@wordpress/core-data/build-types/hooks/use-entity-record';
+import {
+	__experimentalSurface as Surface, // eslint-disable-line
+	__experimentalText as Text, // eslint-disable-line
+} from '@wordpress/components';
+
+function HoverContent() {
+	return (
+		<Surface
+			style={ {
+				margin: '20vh auto',
+				maxWidth: 400,
+				padding: 20,
+			} }
+			variant="primary"
+		>
+			<Text> { __( 'Autogenerated content', 'woocommerce' ) } </Text>
+		</Surface>
+	);
+}
+
+const updateIFrameBackgroundColor = (
+	iframeRef: React.RefObject< HTMLIFrameElement >,
+	isHovered: boolean
+) => {
+	if ( iframeRef?.current?.contentWindow?.document.body ) {
+		iframeRef.current.contentWindow.document.body.style.backgroundColor =
+			isHovered
+				? '#00000059'
+				: iframeRef.current.contentWindow?.document?.bgColor;
+	}
+};
+
+const getEmailType = ( value: string ) => {
+	return window.WooCommerceEmailEditor.email_types.find(
+		( email_type ) => email_type.value === value
+	)?.id;
+};
+
+const updateiFrameSource = (
+	iframeRef: React.RefObject< HTMLIFrameElement >,
+	url: string
+) => {
+	// Update iframe src using replace to avoid polluting browser history
+	iframeRef?.current?.contentWindow?.location.replace( url );
+};
+
+const DEFAULT_EMAIL_TYPE = 'WC_Email_Customer_Processing_Order';
+
+export function WooContentPlaceholderEditContent() {
+	const { current_post_id, current_post_type } =
+		window.WooCommerceEmailEditor;
+	const { editedRecord, hasResolved } = useEntityRecord(
+		'postType',
+		current_post_type,
+		current_post_id
+	) as EntityRecordResolution< Post >;
+	const { slug: postSlug } = editedRecord;
+	const iframeRef = useRef< HTMLIFrameElement | null >( null );
+	const [ isHovered, setIsHovered ] = useState( false );
+
+	const previewUrlBase = window.WooCommerceEmailEditor?.block_preview_url;
+
+	useEffect( () => {
+		if ( ! hasResolved ) {
+			return;
+		}
+
+		// current email type
+		const currentEmailType = getEmailType( postSlug || '' );
+		if ( currentEmailType && iframeRef.current ) {
+			updateiFrameSource(
+				iframeRef,
+				`${ previewUrlBase }&type=${ currentEmailType }`
+			);
+		}
+	}, [ hasResolved, postSlug, iframeRef, previewUrlBase ] );
+
+	return (
+		<div
+			style={ {
+				position: 'relative',
+			} }
+		>
+			<iframe
+				style={ {
+					width: '100%',
+					height:
+						iframeRef?.current?.contentWindow?.outerHeight ||
```

Comment:
I am not sure about the iframe height.
When I tried to use, for example, a content for the reset password email. It does not look good. But maybe we are okay with that because it is only a temporary solution. What do you think?
<img width="678" alt="Screenshot 2025-03-17 at 15 21 00" src="https://github.com/user-attachments/assets/832c634a-69af-42f9-98b5-3c368ee65030" />